lib/transaction/payload/proregtxpayload.js

```diff
@@ -0,0 +1,202 @@
+ * @property {string} inputsHash	uint256	32	Hash of all the outpoints of the transaction inputs
+ * @property {string} payloadSig	vector	Variable	Signature of the hash of the ProTx fields. Signed with KeyIdOwner
+ */
+function ProRegTxPayload(options) {
+  AbstractPayload.call(this);
+  this.version = CURRENT_PAYLOAD_VERSION;
+
+  if (options) {
+    this.type = options.type
+    this.mode = options.mode
+    this.collateralIndex = options.collateralIndex
+    this.ipAddress = options.ipAddress
+    this.port = options.port
+    this.KeyIdOwner = options.KeyIdOwner
+    this.KeyIdOperator = options.KeyIdOperator
+    this.KeyIdVoting = options.KeyIdVoting
+    this.operatorReward = options.operatorReward
+    this.scriptPayout = options.scriptPayout
+    this.inputsHash = options.inputsHash
+    this.payloadSig = options.payloadSig
+  }
+}
+
+ProRegTxPayload.prototype = Object.create(AbstractPayload.prototype);
+ProRegTxPayload.prototype.constructor = AbstractPayload;
+
+/* Static methods */
+
+/**
+ * Parse raw payload
+ * @param {Buffer} rawPayload
+ * @return {ProRegTxPayload}
+ */
+ProRegTxPayload.fromBuffer = function fromBuffer(rawPayload) {
+  var payloadBufferReader = new BufferReader(rawPayload);
+  var payload = new ProRegTxPayload();
+
+  payload.version = payloadBufferReader.readUInt16LE();
+  payload.type = payloadBufferReader.readUInt16LE();
+  payload.mode = payloadBufferReader.readUInt16LE();
+  payload.collateralIndex = payloadBufferReader.readUInt32LE();
+
+  //ipAddress (todo: might need to change to ipv4)
+  var ipArr = [];
+  for (var i = 0; i < 8; i++) {
+    ipArr.push(payloadBufferReader.read(2).toString('hex'));
+  }
+
+  payload.ipAddress = ipArr.join(':');
+
+  payload.port = payloadBufferReader.readUInt16BE();
+
+  // TODO: not sure about a byte order
+  payload.KeyIdOwner = payloadBufferReader.read(constants.PUBKEY_ID_SIZE).toString('hex');
+  payload.KeyIdOperator = payloadBufferReader.read(constants.PUBKEY_ID_SIZE).toString('hex');
+  payload.KeyIdVoting = payloadBufferReader.read(constants.PUBKEY_ID_SIZE).toString('hex');
+  payload.operatorReward = payloadBufferReader.readUInt16LE();
+
+  var scriptPayoutSize = payloadBufferReader.readVarintNum();
+  payload.scriptPayout = payloadBufferReader.read(scriptPayoutSize).toString('hex');
+
+  payload.inputsHash = payloadBufferReader.read(constants.SHA256_HASH_SIZE).toString('hex');
+
+  var payloadSigSize = payloadBufferReader.readVarintNum();
+
+  if (payloadSigSize > 0) {
+    payload.payloadSig = payloadBufferReader.read(payloadSigSize).toString('hex');
+  }
+
+  if (!payloadBufferReader.finished()) {
+    throw new Error('Unexpected payload size');
+  }
+
+  return payload;
+};
+
+/**
+ * Create new instance of payload from JSON
+ * @param {string|BlockchainUserPayloadJSON} payloadJson
+ * @return {SubTxRegisterPayload}
+ */
+ProRegTxPayload.prototype.fromJSON = function fromJSON(payloadJson) {
```

Comment:
`proRegTxPayload.prototype.fromJSON` should come under `/* Instance methods */`

Reply:
as @antouhou pointed out ^ it should be static but the declaration was incorrect, now corrected